src/query.rs

```diff
@@ -89,6 +89,32 @@ impl Query {
         })
     }
 
+    /// Construct a Tantivy's TermSetQuery
+    #[staticmethod]
+    #[pyo3(signature = (schema, field_name, field_values))]
+    pub(crate) fn term_set_query(
+        schema: &Schema,
+        field_name: &str,
+        field_values: Vec<&PyAny>,
+    ) -> PyResult<Query> {
+        let mut terms: Vec<tv::Term> = Vec::new();
```

Comment:
Since we know how long `field_values` is, perhaps we can use [with_capacity](https://doc.rust-lang.org/std/vec/struct.Vec.html#method.with_capacity)?  Then we won't need multiple allocations in the loop.  Just an idea.

Reply:
Yes, makes total sense. I wasn't aware of this API. I just updated and rebased the PR with the latest master.

Reply:
Sorry for being late to the party. More often than not, it is preferable to use the built-in iterator adaptors with automatically do things like pre-allocate collections if possible, i.e. in this case collect should achieve the same effect but avoid adding the loop and allocation noise.

Reply:
Specifically, something like

```rust
let terms = field_values
  .into_iter()
  .map(|field_value| {
    make_term(&schema.inner, field_name, &field_value).map_err(|err| {
      exceptions::PyTypeError::new_err(format!(
          "Cannot create a term from the value: {}",
          field_value
      ))
    })
  })
  .collect::<Result<Vec<_>, _>>()?;
```

should work and be just as efficient.

Reply:
Much better. Thanks for going the extra mile!